File: cpp/Array and Hashing/3_twoSum.cpp
//take a hashmap and check if complement of a no already exists. If it does return if not store the number to that index.

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> m;
        int n = nums.size();

        for(int i =0;i<n;i++){
            int complement = target - nums[i];
            if(m.count(complement)){
                return {m[complement],i};
            }
            m[nums[i]] = i;
        }
        return {};
    }
};

File: Python/Array and Hashing/3_twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        prevMap = {} #value : index
        for i,n in enumerate(nums):
            complement = target - n
            if complement in prevMap:
                return [prevMap[complement], i]
            prevMap[n] = i
        return
        
